src/components/pages/Cart.js

```diff
@@ -1,9 +1,111 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { Link } from 'react-router-dom';
+import { CartContext } from '../../contexts/CartContext';
 
 const Cart = () => {
+  const { cart, setCart } = useContext(CartContext);
+  // getting total cost of items * quantity
+  const costTimesQuantityTotals = [];
+
+  for (let i = 0; i < cart.length; i++) {
+    costTimesQuantityTotals.push(cart[i].quantity * cart[i].price);
+  }
+
+  // checking out resets the cart
+  const handleCheckout = () => {
+    setCart([]);
+  };
+
+  // runs when clicking 'X' button
+  const removeItem = (name) => {
+    setCart(cart.filter((item) => item.name !== name));
+  };
+
+  // runs when changing item quantity
+  const changeQuantity = (e, name) => {
+    setCart([
+      ...cart,
+      cart.map((item) => {
+        if (item.name === name) {
+          // checking to make sure the value doesn't exceed the number in stock (item.max)
+          if (e.target.value > item.max) {
+            return (e.target.value = item.max - 1);
+          }
+          // if the check passes, the input value gets added to the quantity number
+          return (item.quantity = +e.target.value);
+        } else {
+          return cart;
+        }
+      }),
+    ]);
+    // if the quantity for any item is less than 1, remove it from the cart list
+    setCart(cart.filter((item) => item.quantity > 0));
```

Comment:
You can use `cart.reduce` instead of `cart.map` to include this logic. That way you don't have to iterate over your list twice.